### DISCUSSION TITLE
Why the Queue can  not bind the routing key in kombu 4.6.11 ?

### DISCUSSION THREAD

**Discussion by honghh2018** [celery/kombu#2324] [Q&A]
Hi developer,
     
     the queue name was normal_queue for NORMAL_QUEUE_NAME
    the NORMAL_ROUTING_KEY value was normal_routing_key 
    but the web show the normal_queue had not the routing key
    
![image](https://github.com/user-attachments/assets/9dd60323-be01-42f4-a56d-4ebb84acab3c)

    
    the code lying below:
     Queue(RabbitNameConfig.NORMAL_QUEUE_NAME,
                                  self.exchange_normal,
                                  routing_key=RabbitNameConfig.NORMAL_ROUTING_KEY,
                                  arguments={
                                      'x-dead-letter-exchange': RabbitNameConfig.DLX_EXCHANGE_NAME,  
                                      'x-dead-letter-routing-key': RabbitNameConfig.DLX_ROUTING_KEY,  
                                      # 'x-message-ttl':10000   # 
                                      'x-message-ttl': int(
                                          USER_TRANS_LIMIT_CLASS.TRANS_UNPAY_TIMEOUT_CLOSE_SECOND * 1000)
                                     
                                  })
                                  
                                  
 Is it the Queue can not bind the exchange and routing key?
 and the version 4.6.11 can set the arguements with 
   arguments={
                                      'x-dead-letter-exchange': RabbitNameConfig.DLX_EXCHANGE_NAME,  
                                      'x-dead-letter-routing-key': RabbitNameConfig.DLX_ROUTING_KEY,  
                                      # 'x-message-ttl':10000   # 
                                      'x-message-ttl': int(
                                          USER_TRANS_LIMIT_CLASS.TRANS_UNPAY_TIMEOUT_CLOSE_SECOND * 1000)
                                     
                                  }
 ?
 Any advice would be appreciated
 Best,
hanhuihong

**Comment by TayfurYldz** [comment]
A `Queue` with an exchange and `routing_key` does create a binding when the queue is actually declared through Kombu. In the current implementation, `Queue.declare()` performs `queue_declare()` and then `queue_bind()`, and that bind uses the queue's `routing_key`.

So the constructor shown here is not enough by itself to prove that RabbitMQ has received the binding. Check that this exact `Queue` object is being declared/consumed (for example `queue.declare()` on the active connection/channel), and check the exchange name/type in RabbitMQ rather than only the queue declaration.

The `arguments` dictionary is separate from routing: `x-dead-letter-exchange`, `x-dead-letter-routing-key`, and `x-message-ttl` are queue declaration arguments and are supported independently of the normal exchange→queue binding key. If the queue exists but the binding is absent, the useful next diagnostic is the code that declares/consumes this `Queue`, not changing those arguments.
